File: algorithm_study_java/src/baekjoon_online/p_1110/p_1110.java
package baekjoon_online.p_1110;

import java.util.Scanner;

/*
Input:
26

Output:
4
 */

public class p_1110 {
    public static void main(String[] args) {
        // First line
        Scanner sc = new Scanner(System.in);
        int givenNumber = sc.nextInt();

        int count = 0;
        int generated = givenNumber;

        while(true) {
            count++;
            int firstNumber = generated / 10;
            int secondNumber = generated % 10;
            generated = (secondNumber * 10) + ((firstNumber + secondNumber) % 10);

            if (generated == givenNumber) {
                break;
            }
        }
        System.out.println(count);
    }
}

File: algorithm_study_python/baekjoon_online/p_1110/p_1110.py
'''
Input:
26

Output:
4

'''
input_number = int(input())

generated_number = input_number
count = 0
while True:
    first_number = generated_number//10
    second_number = generated_number%10
    generated_number = (second_number*10) + ((first_number + second_number)%10)
    # print("generated_number=", generated_number)
    count += 1

    if generated_number == input_number :
        break;

print(count)
